packages/controllers/src/utils.ts

```diff
@@ -52,6 +52,71 @@ export function isValidJson(value: unknown): value is Json {
   }
 }
 
+export function objectDiff<A, B>(a: A, b: B): Diff<A, B> {
+  return Object.entries(a).reduce((acc, [key, value]) => {
+    if (key in b) {
+      return acc;
+    }
+    acc[key] = value;
+    return acc;
```

Comment:
All of this could be simplified to a simple arrow function
```
=> key in b ? acc : ({ ...acc, [key]: value })
```

Reply:
Clever, but this creates a new object a has to iterate on all keys on every iteration, making it into O(n^2) algorithm. Compared to reused single object in the current one with O(n) complexity.

Reply:
I guess it depends on if we value readability over performance. Not sure what our standards are for that kind of thing. 

Either way if you want to keep this I would simplify to
```
if(!(key in b)){
  acc[key] = value;
}
return acc;
```